### DISCUSSION TITLE
List of all indicators / table of content

### DISCUSSION THREAD

**Discussion by christoph-LSN** [open-sdg/open-sdg#2005] [Ideas]
Hello @MoBosse, @brockfanning @sarah-beasley, 

I would like to have a page listing all indicators. Something like the reportingstatus (https://christoph-lsn.github.io/MT_Site/reporting-status/) but listing all indicators for all goals. Do you have an idea how to realize this?

site: https://christoph-lsn.github.io/MT_Site/
data repo: https://github.com/christoph-LSN/IM_Data
site repo: https://github.com/christoph-LSN/MT_Site

**Comment by sarah-beasley** [comment]
Hi @christoph-LSN, can I just check - do you want the reporting status functionality for each indicator too?

**Reply by christoph-LSN** [reply]
@sarah-beasley Not, yet. It's more a kind of table of contents I need.

**Comment by brockfanning** [comment]
@christoph-LSN You'll want to use Jekyll template syntax (Liquid) and the Jekyll variables described here: https://open-sdg.readthedocs.io/en/latest/jekyll-templates/#variables-available-on-all-pages

For example, to list all of the goal names as an H2, you would do something like:

```
{% for goal in page.goals %}
  <h2>{{ goal.name }}</h2>
{% endfor %}
```

Similarly, to list all indicator names as an H2:

```
{% for indicator in page.indicators %}
  {{ indicator.name }}
{% endfor %}
```

If you want to have the indicators grouped by goal (ie, each goal name as an H2 and each indicator name as an H3), you could do a loop within a loop, like:

```
{% for goal in page.goals %}
  <h2>{{ goal.name }}</h2>
  {% for indicator in page.indicators %}
    {% if indicator.goal_number == goal.number %}
      <h3>{{ indicator.name }}</h3>
    {% endif %}
  {% endfor %}
{% endfor %}
```

Each indicator object has various variables you can use, as described here: https://open-sdg.readthedocs.io/en/latest/jekyll-templates/#indicator-objects

For example to make the indicator names link to the indicator pages, you could do:

```
<h3>
  <a href="{{ indicator.url }}">{{ indicator.name }}</a>
</h3>
```

**Reply by christoph-LSN** [reply]
@brockfanning great! I'll try this. One Question. I added a file e.g. [indicator_list.html](https://github.com/christoph-LSN/MT_Site/blob/develop/_includes/components/indicator_list.html) (with your suggested code) in the _includes/components folder.

Do I need create a page e.g indicator_list.md in the _pages to folder get the new page on my frontpage between "Themenbereiche" and "Berichtsstatus"?

![image](https://github.com/open-sdg/open-sdg/assets/94443931/59d7b6e4-0fd6-4a27-af41-cf3ce019bc88)

**Comment by christoph-LSN** [comment]
@brockfanning small update: I got ist so far, that all goals and indicators are listet here: https://christoph-lsn.github.io/MT_Site/indicator_list/ (using this file: https://github.com/christoph-LSN/MT_Site/blob/develop/_pages/indicator_list.md)

But I am strugeling with the `<h3> and the <a href="{{ indicator.url }}">{{ indicator.name }}</a>` this won't show correctly

**Comment by MoBosse** [comment]
@christoph-LSN Try renaming the file `indicator_list.md` as `indicator_list.html`...

**Reply by christoph-LSN** [reply]
@MoBosse Thanks, it works....

**Comment by christoph-LSN** [comment]
Thank you @brockfanning an @MoBosse for the quick response!

https://christoph-lsn.github.io/MT_Site/indicator_list/
